src/Nox.Types/Types/Time/Time.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Globalization;
+
+namespace Nox.Types;
+
+/// <summary>
+/// Represents a time of day, as would be read from a clock, within the range 00:00:00 to 23:59:59.9999999.
+/// </summary>
+public sealed class Time : ValueObject<DateTime, Time> // This should be TimeOnly
+{
+    /// <summary>
+    /// MinTimeTicks is the ticks for the midnight time 00:00:00.000 AM
+    /// </summary>
+    private const long MinTimeTicks = 0;
+
+    /// <summary>
+    /// MaxTimeTicks is the max tick value for the time in the day. It is calculated using DateTime.Today.AddTicks(-1).TimeOfDay.Ticks.
+    /// </summary>
+    private const long MaxTimeTicks = 863_999_999_999;
+
+    /// <summary>
+    /// Creates the Time object.
+    /// </summary>
+    /// <param name="hours">The hours.</param>
+    /// <param name="minutes">The minutes.</param>
+    /// <returns>A Time.</returns>
+    public static Time From(int hours, int minutes) => From(hours, minutes, 0, 0);
+
+    /// <summary>
+    /// Creates the Time object.
+    /// </summary>
+    /// <param name="hours">The hours.</param>
+    /// <param name="minutes">The minutes.</param>
+    /// <param name="seconds">The seconds.</param>
+    /// <returns>A Time.</returns>
+    public static Time From(int hours, int minutes, int seconds)=> From(hours, minutes, seconds, 0);
+
+    /// <summary>
+    /// Creates the Time object.
+    /// </summary>
+    /// <param name="hours">The hours.</param>
+    /// <param name="minutes">The minutes.</param>
+    /// <param name="seconds">The seconds.</param>
+    /// <param name="milliseconds">The milliseconds.</param>
+    /// <returns>A Time.</returns>
+    public static Time From(int hours, int minutes, int seconds, int milliseconds)
+    {
+        var timeSpan = new TimeSpan(days: 0, hours: hours, minutes: minutes, seconds: seconds, milliseconds: milliseconds);
+        return From(timeSpan);
+
+    }
+
+    /// <summary>
+    /// Creates the Time object from ticks.
+    /// </summary>
+    /// <param name="ticks">The ticks.</param>
+    /// <returns>A Time.</returns>
+    public static Time From(long ticks)
+    {
+        var timeSpan = new TimeSpan(ticks);
+        return From(timeSpan);
+
+    }
+
+    private static Time From(TimeSpan timeSpan)
+    {
+        var newObject = new Time();
+
+        var result = newObject.Validate();
+        
+        if (timeSpan.Ticks < Time.MinTimeTicks)
```

Comment:
Should this be done on the Validate method?

Reply:
Modified the implementation, this was a validation on intermediate timespan and not on the TValue(DateTime).